server/server/src/static_files.rs

```diff
@@ -232,55 +172,64 @@ async fn upload_sync_file(
 
-    let db_connection = service_provider
-        .connection()
-        .map_err(|err| InternalError::new(err, StatusCode::INTERNAL_SERVER_ERROR))?;
+    let static_file = upload_sync_file_inner(&service_provider, &settings, path.into_inner(), file)
+        .await
+        .map_err(|error| {
+            log::error!("Error while uploading file: {}", format_error(&error));
+            InternalError::new(
+                "Error uploading file, please check server logs",
+                StatusCode::INTERNAL_SERVER_ERROR,
+            )
+        })?;
+
+    Ok(HttpResponse::Ok().json(static_file.id))
+}
 
-    let (table_name, record_id) = path.into_inner();
+#[derive(Error, Debug)]
+enum UploadFileError {
+    #[error("Database error")]
+    DatabaseError(#[from] RepositoryError),
+    #[error("Other")]
+    Other(#[from] anyhow::Error),
+}
 
-    let files = handle_file_upload(
-        payload,
-        settings,
-        StaticFileCategory::SyncFile(table_name.clone(), record_id.clone()),
+async fn upload_sync_file_inner(
+    service_provider: &ServiceProvider,
+    settings: &Settings,
+    (table_name, record_id): (String, String),
+    file: TempFile,
+) -> Result<StaticFile, UploadFileError> {
+    let db_connection = service_provider.connection()?;
+
+    let file_service = StaticFileService::new(&settings.server.base_dir)?;
+    // File is 'moved' need these values for SyncFileReferenceRow
+    let total_bytes = usize_to_i32(file.size);
+    let mime_type = file.content_type.as_ref().map(|mime| mime.to_string());
+
+    let static_file = file_service.move_temp_file(
+        file,
+        &StaticFileCategory::SyncFile(table_name.clone(), record_id.clone()),
         None,
-    )
-    .await?;
+    )?;
 
     let repo = SyncFileReferenceRowRepository::new(&db_connection);
-    for file in files.clone() {
-        let result = repo.upsert_one(&SyncFileReferenceRow {
-            id: file.id,
-            file_name: file.filename,
-            table_name: table_name.clone(),
-            mime_type: file.mime_type,
-            uploaded_bytes: 0, // This is how many bytes are uploaded to the central server
-            total_bytes: file.bytes,
-            created_datetime: chrono::Utc::now().naive_utc(),
-            deleted_datetime: None,
-            record_id: record_id.clone(),
-            status: SyncFileStatus::New,
-            direction: repository::sync_file_reference_row::SyncFileDirection::Upload,
-            ..Default::default()
-        });
-        match result {
-            Ok(_) => {}
-            Err(err) => {
-                log::error!(
-                    "Error saving file reference: {} - DELETING UPLOADED FILES",
-                    err
-                );
-                // delete any files that were uploaded...
-                for file in files {
-                    // File::create is blocking operation, use threadpool
-                    web::block(|| std::fs::remove_file(file.path)).await??;
```

Comment:
This would only happen if we fail to upsert (database error) if there is database error we have a major server problem, and having a stray file shouldn't be too much of an issue

Reply:
Agreed, Mark wanted this in there but I don't think it should happen.
With postgres on a separate server I guess the chance is a bit higher but I don't think it would be a major thing if there are a few extra files on disk in this situation.